### PR TITLE
feat: add obligation records context and related types

### PR DESCRIPTION
**Pull request by manishdex25**
- Introduced a new JSON context for obligation records.
- Updated index.ts to include the new obligation records context.
- Added ObligationRecordsCredentialStatus type to manage obligation records.
- Implemented assertion function for ObligationRecordsCredentialStatus in utils.ts.
- Enhanced credential status checking to accommodate obligation records.

## Summary
What is the background of this pull request?

## Changes
* What are the changes made in this pull request?
* Change this and that, etc...

## Issues
What are the related issues or stories?

## Releases
Channels: latest
ETA: Any target release date

## Summary by CodeRabbit

- **New Features**
  - Added JSON-LD contexts for bills of exchange and obligation records.
  - Added obligation-records credential status support, including obligation registry details.
  - Introduced stricter chainId validation (integer or numeric-string) for these credential statuses.
- **Documentation**
  - Updated context catalog with new URLs and added sample resources.
- **Bug Fixes**
  - Improved credential-status validation/routing when obligation registry details are present.
- **Tests & Chores**
  - Expanded validation test coverage for chainId edge cases and updated context asset staging.

### CODE REVIEW

File: packages/w3c-context/README.md [hunk 1]
```diff
@@ -69,17 +69,25 @@ The package includes the following pre-cached contexts:
...
 - `https://trustvc.io/context/bill-of-lading-carrier.json` - Bill of Lading Carrier
 - `https://trustvc.io/context/coo.json` - Certificate of Origin
 - `https://trustvc.io/context/invoice.json` - Invoice
 - `https://trustvc.io/context/promissory-note.json` - Promissory Note
 - `https://trustvc.io/context/warehouse-receipt.json` - Warehouse Receipt
 
+### Samples
+
+- `samples/obligation-credential-subject.sample.json` - Example Obligation / BoE credential subject payload
+- `samples/bill-of-exchange.md` - Form label → property map for finalized BoE fields
```
[the remark this answers is not in the document]
Comment (manishdex25) [reply packages/w3c-context/README.md:90]: already fixed

**Review by rongquan1 — APPROVED** [review]
